vendor/assets/javascripts/initializers.js

```diff
@@ -54,4 +54,23 @@ $(document).on('turbolinks:load', function() {
       }
     });
   });
+  $('.js-dropzone-featured-image').each(function() {
+    $(this).find('form').dropzone({
+      dictDefaulMessage: 'Drop files or click here to upload',
+      init: function() {
+        this.on('success', function(req, res) {
+          var clone = $($.clone($('.js-featured-image-table-element')[0]));
+          var baseId = clone.find('input').attr('id').split('_');
+          baseId.pop();
+          baseId.push(res.id);
+          var radioButton = clone.find('input');
+          radioButton.attr('id', baseId);
+          radioButton.val(res.id);
+          radioButton.attr('checked', true);
+          clone.insertBefore($('.js-featured-image-table-element')[0]);
+          var featuredImage = $('.js-featured-image').attr('src', res.attachment.url);
```

Comment:
Yeah, I've already had a few thoughts on that but didn't implement them to save on time. I was thinking do something like have a function that you pass into an element that you want to copy and change the id and value of, an element that you want to insert it into the dom before/after and an optional callback to do any custom stuff inside of. There are obviously more details than that but that's the basics of what I'm thinking. I also have some thoughts on refactoring out some common functionality in other pieces of cmsify_assets, mostly surrounding the node inherits pattern.